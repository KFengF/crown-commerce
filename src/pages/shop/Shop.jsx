import React from 'react';
import CollectionPreview from './components/collection-preview/CollectionPreview';
import shopData from './shopData';

class Shop extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            collections: shopData
        }
    }

    render() {
        const { collections } = this.state
        return (
            <div className="shop-page" >
                { collections.map( ({ id, ...otherCollectionAttrs }) => (
                    <CollectionPreview key={ id } { ...otherCollectionAttrs } />
                )) }
            </div>
        )
    }
}

export default Shop;