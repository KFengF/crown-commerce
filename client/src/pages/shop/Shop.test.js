import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import { ShopRouter } from './Shop.router';

describe('Shop router', () => {
    let wrapper;
    let mockFetchCollectionsStart;

    beforeEach(() => {
        mockFetchCollectionsStart = jest.fn();
        const mockProps = {
            collections: null,
            fetchCollectionsStart: mockFetchCollectionsStart,
            match: { path: '/shop' },
            isCollectionsFetching: true
        }
        wrapper = mount(
            <BrowserRouter>
                <ShopRouter { ...mockProps } />
            </BrowserRouter>
        );
    });

    it('should render Shop router', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should call fetchCollectionsStart when mounted', done => {
        setImmediate(() => {
            wrapper.update();
            expect(mockFetchCollectionsStart).toHaveBeenCalled();
            done();
        })
    });
});