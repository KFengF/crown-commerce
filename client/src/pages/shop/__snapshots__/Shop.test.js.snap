// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop router should render Shop router 1`] = `ReactWrapper {}`;
